src/pages/settings/Payments/PaymentMethodList.js

```diff
@@ -54,87 +63,75 @@ const defaultProps = {
+    enableSelection: false,
+    selectedAccountID: '',
+    filterType: '',
     isAddPaymentMenuActive: false,
 };
 
 class PaymentMethodList extends Component {
     constructor(props) {
         super(props);
-
+        this.state = {
+            selectedAccountID: this.props.selectedAccountID,
+        };
         this.renderItem = this.renderItem.bind(this);
+        this.selectPaymentMethod = this.selectPaymentMethod.bind(this);
+        this.createPaymentMethodList = this.createPaymentMethodList.bind(this);
     }
 
     /**
      * Take all of the different payment methods and create a list that can be easily digested by renderItem
-     *
      * @returns {Array}
      */
     createPaymentMethodList() {
-        const combinedPaymentMethods = [];
+        let paymentMethods = [];
+        if (_.isEmpty(this.props.filterType)) {
+            paymentMethods = PaymentUtils.getPaymentMethodsList(this.props.bankAccountList, this.props.cardList, this.props.payPalMeUsername);
+        } else {
+            paymentMethods = PaymentUtils.getPaymentMethodsList(
+                this.props.filterType === CONST.PAYMENT_METHODS.BANK_ACCOUNT ? this.props.bankAccountList : [],
+                this.props.filterType === CONST.PAYMENT_METHODS.DEBIT_CARD ? this.props.cardList : [],
+                !this.props.filterType ? this.props.payPalMeUsername : '',
+            );
+        }
 
-        _.each(this.props.bankAccountList, (bankAccount) => {
-            // Add all bank accounts besides the wallet
-            if (bankAccount.type === CONST.BANK_ACCOUNT_TYPES.WALLET) {
-                return;
-            }
+        const combinedPaymentMethods = _.map(paymentMethods, (method) => {
+            let iconProperties;
 
-            const formattedBankAccountNumber = bankAccount.accountNumber
-                ? `${this.props.translate('paymentMethodList.accountLastFour')} ${
-                    bankAccount.accountNumber.slice(-4)
-                }`
-                : null;
-            const {icon, iconSize} = getBankIcon(lodashGet(bankAccount, 'additionalData.bankName', ''));
-            combinedPaymentMethods.push({
-                type: MENU_ITEM,
-                title: bankAccount.addressName,
-
-                // eslint-disable-next-line
-                description: formattedBankAccountNumber,
-                icon,
-                iconSize,
-                onPress: e => this.props.onPress(e, bankAccount.bankAccountID),
-                key: `bankAccount-${bankAccount.bankAccountID}`,
-            });
-        });
+            switch (method.type) {
+                case CONST.PAYMENT_METHODS.BANK_ACCOUNT: iconProperties = getBankIcon(method.bankName); break;
+                case CONST.PAYMENT_METHODS.DEBIT_CARD: iconProperties = getBankIcon(method.bankName, true); break;
+                case CONST.PAYMENT_METHODS.PAYPAL: iconProperties = {icon: Expensicons.PayPal}; break;
+                default: break;
+            }
```

Comment:
```suggestion
            switch (method.type) {
                case CONST.PAYMENT_METHODS.BANK_ACCOUNT: 
                    iconProperties = getBankIcon(method.bankName); 
                    break;
                case CONST.PAYMENT_METHODS.DEBIT_CARD: 
                    iconProperties = getBankIcon(method.bankName, true); 
                    break;
                case CONST.PAYMENT_METHODS.PAYPAL: 
                    iconProperties = {icon: Expensicons.PayPal}; 
                    break;
                default: break;
            }
```

Reply:
Alternatively, move this to a method and return out of the switch instead of reassigning `iconProperties`.